unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Tests/LoaderShould.cs

```diff
@@ -0,0 +1,500 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using AvatarSystem;
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.Helpers;
+using NSubstitute;
+using NSubstitute.Extensions;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Object = UnityEngine.Object;
+
+namespace Test.AvatarSystem
+{
+    public class LoaderShould
+    {
+        private const string FEMALE_ID = "urn:decentraland:off-chain:base-avatars:BaseFemale";
+        private const string EYES_ID = "urn:decentraland:off-chain:base-avatars:f_eyes_00";
+        private const string EYEBROWS_ID = "urn:decentraland:off-chain:base-avatars:f_eyebrows_00";
+        private const string MOUTH_ID = "urn:decentraland:off-chain:base-avatars:f_mouth_00";
+        private static readonly string[] WEARABLE_IDS = new []
+        {
+            "urn:decentraland:off-chain:base-avatars:black_sun_glasses",
+            "urn:decentraland:off-chain:base-avatars:bear_slippers",
+            "urn:decentraland:off-chain:base-avatars:hair_anime_01",
+            "urn:decentraland:off-chain:base-avatars:f_african_leggins",
+            "urn:decentraland:off-chain:base-avatars:blue_bandana",
+            "urn:decentraland:off-chain:base-avatars:bee_t_shirt"
+        };
+
+        private IWearableLoaderFactory wearableLoaderFactory;
+        private Loader loader;
+        private GameObject container;
+        private IAvatarMeshCombinerHelper meshCombiner;
+
+        [SetUp]
+        public void SetUp()
+        {
+            container = new GameObject("Container");
+            PrepareCatalog();
+
+            meshCombiner = Substitute.For<IAvatarMeshCombinerHelper>();
+            wearableLoaderFactory = Substitute.For<IWearableLoaderFactory>();
+            loader = new Loader(wearableLoaderFactory, container, meshCombiner);
+        }
+
+        private void PrepareCatalog()
+        {
+            //This is really, really ugly. There's no other way to solve it until the catalog is in our service locator
+            container.AddComponent<CatalogController>();
+            AvatarAssetsTestHelpers.CreateTestCatalogLocal();
+        }
+
+        [Test]
+        public void GetNewLoaders_WhenEmptyCurrentLoaders()
```

Comment:
snake case test names, could you convert to: `AllAreNewLoadersWhenThereWasntAnyCreatedBefore`